### DISCUSSION TITLE
How do I use trillium-aws-lambda?

### DISCUSSION THREAD

**Discussion by mathiversen** [trillium-rs/trillium#204] [Q&A]
Hello,

I'm struggling a bit to understand how build and deploy on aws, and I'm not sure if I'm doing something wrong or there's in fact a bug preventing the runtime from executing. In short I do the following:

1. I add `x86_64-unknown-linux-musl` to rustup & installed `musl-gcc`
2. I successfully build with the above target
3. I name the executable `bootstrap` and zip it (and make sure the only file is the build file called bootstrap)
4. I login to aws and create a function, I've tried both of the custom runtimes but I get the same error;

```json
"errorMessage":"RequestId: xxx Error: fork/exec /var/task/bootstrap: no such file or directory",
"errorType":"Runtime.InvalidEntrypoint"
```
I started on a tutorial just to document what i was doing: [link](https://github.com/mathiversen/trillium/commit/b0e771fd69ce36e8d6da9c767cf1261b13daf982) maybe that's something worth adding to the project?

**Comment by jbr** [comment]
It has been a bit since I have tried this myself, but [this deploy script](https://github.com/trillium-rs/trillium/blob/main/aws-lambda-example/deploy) is how I was building and deploying the aws-lambda-example app, but I'm not sure if documenting lambda best practices is really in scope for this project, as there are many valid workflows for lambda, and the parts that are unique to trillium end at building a binary

**Reply by mathiversen** [reply]
Yeah i saw the script and that should probably be enough for most people. I just walked through the process to try and figure out what i was doing wrong. I saw that you made some updates to the code, did you try and run it on aws as well?

**Reply by mathiversen** [reply]
Seems like aws have an [official runtime](https://github.com/awslabs/aws-lambda-rust-runtime) so it's probably a good idea to replace  `lamedh_runtime` with `lambda_runtime`. @jbr

**Reply by mathiversen** [reply]
I managed to build working lambda executable with [cross](https://github.com/cross-rs/cross). But when I tried it with trillium i got the following error: 
```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error("data did not match any variant of untagged enum
LambdaRequest", line: 0, column: 0)', /cargo/registry/src/github.com-1ecc6299db9ec823/trillium-aws-lambda-0.2.0/src/lib.rs:88:10
```

**Reply by mathiversen** [reply]
I took a stab at creating my own aws_lambda based on the official runtime. It's not working yet, and I'm trying to debug the last step. It's a pretty good learning opportunity. Feel free to have a look: https://github.com/mathiversen/sessile-aws-lambda @jbr
